app/src/main/java/org/standardnotes/notes/LoginActivity.kt

```diff
@@ -72,40 +74,117 @@ class LoginActivity : AppCompatActivity() {
                             }
                         })
-                        .setView(input)
+                        .setView(layout)
                         .show()
+                dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE)
+                val okbutton = dialog.getButton(AlertDialog.BUTTON_POSITIVE)
+                input.addTextChangedListener(object : TextWatcher {
+                    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                        //
+                    }
+
+                    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                        okbutton.isEnabled = !input.text.isBlank()
+                    }
 
+                    override fun afterTextChanged(s: Editable?) {
+                        //
+                    }
+                })
+                input.setOnEditorActionListener(OnEditorActionListener { v, actionId, event ->
+                    if (actionId == EditorInfo.IME_ACTION_DONE && okbutton.isEnabled) {
+                        okbutton.performClick()
+                        return@OnEditorActionListener true
+                    }
+                    false
+                })
+                okbutton.isEnabled = false
             } catch (e: Exception) {
                 e.printStackTrace()
                 Toast.makeText(this@LoginActivity, getString(R.string.error_login), Toast.LENGTH_LONG).show()
-                login_progress.visibility = View.GONE
+                hideProgress()
             }
         }
+
+        password.setOnEditorActionListener(OnEditorActionListener { v, actionId, event ->
+            if (actionId == EditorInfo.IME_ACTION_DONE && email_sign_in_button.isEnabled) {
+                email_sign_in_button.performClick()
+                return@OnEditorActionListener true
+            }
+            false
+        })
+
+        val textWatcher = object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                //
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                checkValidInput()
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                //
+            }
+        }
+        server.addTextChangedListener(textWatcher)
+        email.addTextChangedListener(textWatcher)
+        password.addTextChangedListener(textWatcher)
+
+        server.setText(ValueStore(this).server)
+        email.requestFocus()
     }
+
+    private fun checkValidInput() {
+        val valid: Boolean = !server.text.isBlank() && !email.text.isBlank() && !password.text.isBlank()
+        email_sign_in_button.isEnabled = valid
+        sign_up.isEnabled = valid
+    }
+
+    private fun showProgress() {
+        // TODO lock orientation changes?
```

Comment:
I should have been more explicit, but the context in which this TODO was written gives some help.  I don't think we should lock orientation changes, but there is a bug where if you change the orientation while a network call is being made, the ProgressBar goes away (since the Activity was destroyed and recreated.

Locking orientation during login network calls for this is a simple, naive solution, but more likely we'd want to fix the issue in some other way.